js/ui/appDisplay.js

```diff
@@ -362,10 +362,20 @@ const EndlessApplicationView = new Lang.Class({
         return false;
     },
 
+    _reloadIcons: function() {
+        for (let icon of this._allIcons) {
+            if (icon.app) {
+                icon.app.reload_app_info();
+                icon.icon.reloadIcon();
+            }
+        }
+    },
+
     addIcons: function(isHidden) {
         // Don't do anything if we don't have more up-to-date information, since
         // re-adding icons unnecessarily can cause UX problems
         if (!this.iconsNeedRedraw()) {
+            this._reloadIcons();
```

Comment:
Okay. That makes sense. I thought it'd not be that problematic, but it definitely doesn't scale so I will go for a smarter approach.